tests/unit/dense_eWiseMul.cpp

```diff
@@ -0,0 +1,236 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include "../utils/print_alp_containers.hpp"
+
+#include <alp.hpp>
+
+typedef int T;
+constexpr T ALPHA_VALUE = 3;
+constexpr T BETA_VALUE = 2;
+
+// Function used to initialize elements of matrix A
+T f_A( const size_t i, const size_t j, const size_t n ){
+	return static_cast< T >( n * i + j );
+}
+
+// Function used to initialize elements of matrix B
+T f_B( const size_t i, const size_t j, const size_t n ){
+	return static_cast< T >( i + n * j );
+}
+
+// Function used to initialize elements of matrix C
+T f_C( const size_t i, const size_t j, const size_t n ){
+	(void)n;
+	return static_cast< T >( i ) - static_cast< T >( j );
+}
+
+// Function used to calculate C = alpha .* B
+T f_alphaB( const size_t i, const size_t j, const size_t n ) {
+	const T b_value = f_B( i, j, n );
+	const T c_value = f_C( i, j, n );
+	return c_value + ALPHA_VALUE * b_value;
+}
+
+// Function used to calculate C = A .* beta
+T f_Abeta( const size_t i, const size_t j, const size_t n ) {
+	const T a_value = f_A( i, j, n );
+	const T c_value = f_C( i, j, n );
+	return c_value + a_value * BETA_VALUE;
+}
+
+// Function used to calculate C = A .* B
+T f_AB( const size_t i, const size_t j, const size_t n ) {
+	const T a_value = f_A( i, j, n );
+	const T b_value = f_B( i, j, n );
+	const T c_value = f_C( i, j, n );
+	return c_value + a_value * b_value;
+}
+
+template< typename MatrixType, typename Function >
+alp::RC initialize( MatrixType &A, const Function &f ) {
+	const size_t n = nrows( A );
+	const alp::RC rc = alp::eWiseLambda(
+		[ &f, n ]( const size_t i, const size_t j, T &val ) {
+			val = f( i, j, n );
+		},
+		A
+	);
+
+	if( rc != alp::SUCCESS ){
+		std::cerr << "\tFailed to initialize matrix. alp::eWiseLambda (matrix, no vectors) FAILED\n";
+	}
+	return rc;
+}
+
+// Checks if each element of provided matrix match the value calculated by the provided function
```

Comment:
perhaps to have version or check_correctness for matrices, vectors and scalars and to have them in utils?